<?xml version="1.0" ?>

<launch>
    <!-- (start your robot's MoveIt! stack, e.g. include its moveit_planning_execution.launch) -->
    <!-- (start your tracking system's ROS driver) -->
    <arg name="camera_rgb_topic" default="/camera/color/image_raw" />
    <arg name="camera_info_topic" default="/camera/color/camera_info" />
    <arg name="camera_frame_name" default="camera_color_optical_frame" />

    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" >
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(arg camera_rgb_topic)" />
        <remap from="camera_info" to="$(arg camera_info_topic)" />
        <param name="camera_frame" type="str" value="$(arg camera_frame_name)" />
        <param name="publish_tag_detections_image" type="bool" value="true" />
        <rosparam command="load" file="$(find easy_handeye)/config/apriltag/settings.yaml"/>
        <rosparam command="load" file="$(find easy_handeye)/config/apriltag/tags.yaml"/>
    </node>

    <include file="$(find easy_handeye)/launch/calibrate.launch">
        <arg name="eye_on_hand" value="true"/>

        <!-- you can choose any identifier, as long as you use the same for publishing the calibration -->
        <!--    <arg name="namespace" value="pr2_handeye_calib"/> -->

        <!-- fill in the following parameters according to your robot's published tf frames -->
        <arg name="robot_base_frame" value="base_link"/>
        <arg name="move_group" value="left_arm"/>
        <arg name="robot_effector_frame" value="l_gripper_tool_frame"/>

        <!-- fill in the following parameters according to your tracking system's published tf frames -->

        <arg name="tracking_base_frame" value="$(arg camera_frame_name)"/>
        <arg name="tracking_marker_frame" value="tag_5"/>

    </include>
</launch>

